model/map/src/main/java/org/keycloak/models/map/storage/chm/ConcurrentHashMapStorage.java

```diff
@@ -88,22 +93,35 @@ public long delete(ModelCriteriaBuilder<M> criteria) {
         }
         Predicate<? super K> keyFilter = b.getKeyFilter();
         Predicate<? super V> entityFilter = b.getEntityFilter();
-        res = 0;
-        for (Iterator<Entry<K, V>> iterator = store.entrySet().iterator(); iterator.hasNext();) {
-            Entry<K, V> next = iterator.next();
-            if (keyFilter.test(next.getKey()) && entityFilter.test(next.getValue())) {
-                res++;
-                iterator.remove();
-            }
+        Stream<Entry<K, V>> storeStream = store.entrySet().stream();
+        final AtomicLong res = new AtomicLong(0);
+
+        if (queryParameters != null && queryParameters.getOrdering() != null && !queryParameters.getOrdering().isEmpty()) {
```

Comment:
Would it be possible to avoid all of these `null` checks?

Reply:
It seems that this question is related to the first one regarding the default instance of `QueryParameters`. The question is whether we want to have some default comparator for each entity or we want to allow pagination without setting the `orderBy` attribute and rely only on the storage that each call will result in the same entity ordering and hence the pagination will be correct.

Reply:
Exactly:
- There should be a default pagination with a given maximum number of objects returned by a query, say 100 objects.
- The default order would be empty.
The providers that leverage this pagination should set the order explicitly if there should be one.